### PR TITLE
Create Naive Estimator

### PR DESCRIPTION
**Pull request by ganesh-krishnan**
## What
- Create a naive estimator. A naive estimator is one that just returns the mean of the outcome variable.
- Support `predict_proba` on models if the underlying estimator allows it
- Add `predict_proba` method to xgboost to access probabilities

## Why
1. Naive models are useful for benchmarking
2. We should be able to access probabilities from classifiers when available

### CODE REVIEW

**Comment by hose314** [comment]
> We should be able to access probabilities from classifiers when available
+1

File: lore/estimators/naive.py [hunk 1]
```diff
@@ -1,3 +1,11 @@
+# -*- coding: utf-8 -*-
+"""
+Naive Estimator
+****************
+A naive estimator is a useful baseline against which to benchmark more complex models.
+A naive estimator will return the mean of the outcome for regression models and
+the majority class for classification models.
```
Comment (montanalow) [inline lore/estimators/naive.py]: majority or plurality?

File: lore/estimators/naive.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        frame, filename, line_number, function_name, lines, index = inspect.stack()[1]
+        super(Naive, self).__init__()
+
+
+class Regression(Base):
+    @before_after_callbacks
+    @timed(logging.INFO)
+    def predict(self, dataframe):
+        """See :ref:`Base Estimator for Naive _naive_base_predict`"""
+        return numpy.ones(dataframe.shape[0])*self.mean
```
Comment (montanalow) [inline lore/estimators/naive.py]: https://docs.scipy.org/doc/numpy-1.15.1/reference/generated/numpy.full.html seems more straightforward.

File: lore/estimators/naive.py (hunk 1 above)
Comment (ganesh-krishnan) [reply lore/estimators/naive.py]: Plurality for multi-class. But I've only implement a binary classifier for now

**Pull request merged**